### PR TITLE
[CSL-1430] Make generate-key and read-key

### PR DESCRIPTION
**Pull request by naryl**
It can generate a primary key and display information about a key. Please check if that's the required format:
```
$ stack exec -- cardano-keygen read-key --path something.key
[keygen:INFO:ThreadId 4] Processing command
[keygen:INFO:ThreadId 4] Primary: pub:60ced4ca; address hash: 4d2f0122
[keygen:INFO:ThreadId 4] No wallet set
[keygen:INFO:ThreadId 4] Keys: 
[keygen:INFO:ThreadId 4] Vss PK: PublicKey Point 99593608784699692652544980421310389590078448882169371488268607087261604356744 92169266176351874303430525274811642215745492274510204282518003720616292214609
```
I didn't find a better way to display PVSS's keys.

### CODE REVIEW

**Comment by naryl** [comment]
No. Will do before merging.

**Comment by georgeee** [comment]
`add-key` command from wallet CLI will help you!

File: tools/src/keygen/Main.hs [hunk 1]
```diff
@@ -146,6 +152,37 @@ getAvvmGenesis AvvmStakeOptions {..} = do
 rearrange :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m ()
 rearrange msk = mapM_ rearrangeKeyfile =<< liftIO (glob msk)
 
+genPrimaryKey :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m ()
+genPrimaryKey path = generateKeyfile True Nothing path >> pure ()
```
Comment (volhovM) [inline tools/src/keygen/Main.hs]: `void $ generateKeyfile ...`
And yes, some `logInfo "successfully generated X"` would be good.

File: tools/src/keygen/Main.hs [hunk 2]
```diff
@@ -146,6 +152,37 @@ getAvvmGenesis AvvmStakeOptions {..} = do
...
+                           map (showKeyWithAddressHash . decryptESK) $
+                           view usKeys us)
+    logInfo $ maybe "No vss"
+                    (("Vss PK: " <>) . showPvssKey) $
+                    view usVss us
+
+showKeyWithAddressHash :: SecretKey -> Text
+showKeyWithAddressHash sk = sformat (build%"; address hash: "%build) pk ah
+  where
+    pk = toPublic sk
```
Comment (volhovM) [inline tools/src/keygen/Main.hs]: In general there's no sense to have `pk` and `ah` here because they are trivial and only used once, but it's minor.

Comment (naryl) [inline tools/src/keygen/Main.hs]: pk is used twice. First for output, then to get ah.

**Pull request closed without merging**